extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ArcProcessor.java

```diff
@@ -466,7 +466,7 @@ public ObserverRegistrationPhaseBuildItem registerSyntheticObservers(BeanRegistr
         // Initialize the type -> bean map
         beanRegistrationPhase.getBeanProcessor().getBeanDeployment().initBeanByTypeMap();
 
-        // Register a synthetic bean for each List<?> with qualifier @All
+        // Validate @All List<> injection points
```

Comment:
The code does both, though, right? The loop does validation, and then there's registration too.

Reply:
Ah, in fact we can remove the method completely!

Reply:
If so, that sounds good to me :-)